compiler/rustc_typeck/src/check/method/probe.rs

```diff
@@ -1359,6 +1362,26 @@ impl<'a, 'tcx> ProbeContext<'a, 'tcx> {
         })
     }
 
+    fn should_emit_unstable_name_collision_hint(
+        &self,
+        unstable_candidates: &[(Candidate<'tcx>, Symbol)],
+    ) -> bool {
+        let some_candidates_are_not_collision_safe =
+            unstable_candidates.iter().any(|(candidate, _)| {
+                let stab = self.tcx.lookup_stability(candidate.item.def_id);
+                !matches!(
+                    stab,
+                    Some(Stability {
+                        level: StabilityLevel::Unstable { collision_safe: true, .. },
+                        ..
+                    })
+                )
+            });
```

Comment:
Nit: we evaluating the expensive condition before cheap conditions here (not sure whether this matters in practice).

Reply:
Fixed!